### PR TITLE
Move app token provider to extensibility namespace and clarify use

### PR DESCRIPTION
**Pull request by bgavrilMS**
This contains some breaking changes, but API is not meant to be used except for Azure SDK, so should be fine. Highlighted that.

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs [hunk 1]
```diff
@@ -186,11 +187,6 @@ private static void ValidateTokenProviderResult(TokenProviderResult TokenProvide
             {
                 HandleInvalidExternalValueError(nameof(TokenProviderResult.ExpiresInSeconds));
             }
-
-            if (string.IsNullOrEmpty(TokenProviderResult.TenantId))
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs:190]: Do we not need Tid for caching the CCA response eventhough it is part of the partitionKey?

File: src/client/Microsoft.Identity.Client/Extensibility/ConfidentialClientApplicationBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]  // hide confidential client on mobile
+#endif
+    public static class ConfidentialClientApplicationBuilderExtensions
+    {
+        /// <summary>
+        /// Allows setting a callback which returns an access token, based on the passed-in parameters.
+        /// MSAL will pass in its authentication parameters to the callback and it is expected that the callback
+        /// will construct a <see cref="AppTokenProviderResult"/> and return it to MSAL.
+        /// MSAL will cache the token response the same way it does for other authentication results.
+        /// Note:
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/Extensibility/ConfidentialClientApplicationBuilderExtensions.cs:22]: remove Note:?

**Review by trwalke — APPROVED** [review]
Couple of questions but refactoring looks good otherwise

**Review by pmaytak — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/Extensibility/AppTokenProviderParameters.cs [hunk 3]
```diff
@@ -1,14 +1,10 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
```
Comment (SameerK-MSFT) [inline src/client/Microsoft.Identity.Client/Extensibility/AppTokenProviderParameters.cs:3]: using System;

**Review by SameerK-MSFT — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/Extensibility/AppTokenProviderParameters.cs (hunk 3 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Extensibility/AppTokenProviderParameters.cs:3]: Not sure I understand this comment. I just used the "remove unused usings" refactoring tool in VS.

**Pull request merged**